### ISSUE TITLE
Refactor the code to use the equals expressions that evaluate the method call to Kind consistently

### ISSUE THREAD

**Issue by reudismam** [dotnet/roslyn#6005]
I would like to refactor the Roslyn code to use consistently the equals expressions that evaluates the method call to Kind. For instance, the attached snapshot of the class SyntaxTreeExtensions.cs presents in the same if statement the code pattern "**Kind() == SyntaxKind.***" and "IsKind(SyntaxKind.*)". I would like to refactor the code to use only the pattern "**IsKind(SyntaxKind.*)**".
![csharpkind](https://cloud.githubusercontent.com/assets/1970407/10522743/4aef90c4-734d-11e5-90fa-e1d96c91f0e9.png)

**Comment by gafter** [comment]
I don't even know where to classify this. Besides not knowing what component to assign it to, it isn't really actionable by us.

**Comment by bkoelman** [comment]
@reudismam Note that there's a semantic difference: `.IsKind()` performs a null check. So, in your first example, `token.Parent` can be `null`, while `token` cannot.

By changing all to `.IsKind()`, the "null-is-not-expected-here" information is lost.

**Comment by reudismam** [comment]
Thanks, @bkoelman! This is true. However, if token is null, the code will throw a null pointer exception when evaluating token.Parent in the second condition. Do you think it is relevant to have this "null-is-not-expected-here"  information in this context? I noticed that the Roslyn team performed this change in many other locations, even when you have token.kind(). For instance, in the current code of this class, we have:

```
if (token.IsKind(SyntaxKind.DotToken))
```

I thought it would be good to change these locations to keep the consistence, especially in these scenarios, where in a single if statement the two ways are being used, which is kind of confusing.  What do you thing? If you want, I can make this change component by component so that you can evaluate one by one whether it makes sense. @gafter, with respect to performance, the only additional check it does is the null check.

**Comment by bkoelman** [comment]
It's not confusing to me that there are two ways to inspect `Kind` (with and without a null check). Instead, they help me to better understand the code regarding where `null`s are (un)expected to appear.

**Comment by reudismam** [comment]
Thanks @bkoelman! This is a good point. However, I noticed the SyntaxToken.IsKind method does not have a null check. Therefore, the semantics will be the equivalent after the change. The token.Parent.IsKind() does have the null check, but it refers to another method: the SyntaxNode.IsKind. In fact, there are two implementations for the SyntaxToken.IsKind method:

One implementation in the class CSharpExtensions (namespace Microsoft.CodeAnalysis)
```CSharp
public static bool IsKind(this SyntaxToken token, SyntaxKind kind)
        {
            return token.RawKind == (int)kind;
        }
```
And one in the class SyntaxTokenExtensions (namespace Microsoft.CodeAnalysis.CSharp.Extensions

```CSharp
public static bool IsKind(this SyntaxToken token, params SyntaxKind[] kinds)
        {
            return kinds.Contains(token.Kind());
        }

public static bool Contains(this SyntaxKind[] kinds, SyntaxKind kind)
        {
            foreach (var k in kinds)
            {
                if (k == kind)
                {
                    return true;
                }
            }

            return false;
        }
```
The first implementation is more efficient answering @gafter's question, while the second one accept more than one kind as argument.

**Comment by CyrusNajmabadi** [comment]
Feel free to make this change in the Services layer and upwards.  I'm totally fine with the code being made consistent.

**Comment by reudismam** [comment]
Thank you! I will prepare a pull request.

**Comment by davkean** [comment]
Resolved with #6938.
